src/generic-container.ts

```diff
@@ -29,20 +30,52 @@ import {
 import { RandomUuid, Uuid } from "./uuid";
 import { HostPortWaitStrategy, WaitStrategy } from "./wait-strategy";
 
-export class GenericContainer implements TestContainer {
-  public static async fromDockerfile(
-    context: BuildContext,
-    uuid: Uuid = new RandomUuid(),
-    dockerClientFactory: DockerClientFactory = new DockerodeClientFactory()
-  ): Promise<GenericContainer> {
-    const image = uuid.nextUuid();
-    const tag = uuid.nextUuid();
+export class GenericContainerBuilder {
+  private factory: DockerClientFactory;
+  private uuid: Uuid;
+  private buildArgs: BuildArgs;
+
+  constructor(private readonly context: BuildContext) {
```

Comment:
Change constructor to:
```typescript
constructor(
  private readonly context: BuildContext,
  private readonly factory: DockerClientFactory = new DockerodeClientFactory(),
  private readonly uuid: Uuid = new RandomUuid()
) {}
```
And remove the fields

Reply:
But since we already have the API break because of the `.build()` I think it is the best way to only have required parameters in the constructor and pass the others the same way as the new parameters?!

Reply:
Providing defaults makes them optional.

Reply:
okay done